node/src/tests.rs

```diff
@@ -119,3 +119,133 @@ fn purge_chain_purges_relay_and_para() {
 			.exists());
 	}
 }
+
+#[test]
+#[cfg(unix)]
+fn builds_specs_based_on_mnemonic() {
+	use serde_json::json;
+
+	let output = Command::new(cargo_bin("moonbeam"))
+		.arg("build-spec")
+		.arg("--dev")
+		.arg("--mnemonic")
+		.arg("myself dutch allow coast planet high glow parrot parent choice identify match")
+		.arg("--accounts")
+		.arg("3")
+		.output()
+		.unwrap();
+
+	// Gather output as json
+	let chain_spec: serde_json::Value = serde_json::from_slice(output.stdout.as_slice()).unwrap();
+	let expected = json!([
+		[
+			json!("0x3d5bd6a54d5f5292b9fb914db40cd5f7c5540f80"),
+			json!(1208925819614629200000000.0)
+		],
+		[
+			json!("0x8055e8c75a862c0da765ed0848366ef4dc492b33"),
+			json!(1208925819614629200000000.0)
+		],
+		[
+			json!("0x764d008debe9493d851d0476ccba9ec23817e2c9"),
+			json!(1208925819614629200000000.0)
+		],
+		// This is Geralds, which is also added
+		[
+			json!("0x6be02d1d3665660d22ff9624b7be0551ee1ac91b"),
+			json!(1208925819614629200000000.0)
+		]
+	]);
+
+	assert_eq!(
+		chain_spec["genesis"]["runtime"]["palletBalances"]["balances"]
+			.as_array()
+			.unwrap(),
+		expected.as_array().unwrap()
+	);
+}
+
+#[test]
+#[cfg(unix)]
+fn export_genesis_state() {
+	let output = Command::new(cargo_bin("moonbeam"))
+		.arg("export-genesis-state")
+		.arg("--chain")
+		.arg("local")
+		.output()
+		.unwrap();
+
+	let expected = "307830303030303030303030303030303030303030303030303030303030303030303030\
+	3030303030303030303030303030303030303030303030303030303030303030303865323533313462316534313533\
+	3539353039623063666566363664343063316437663030666237393637623461386330663231336135626634633130\
+	3033303331373061326537353937623762376533643834633035333931643133396136326231353765373837383664\
+	386330383266323964636634633131313331343030";
+
```

Comment:
This failed for me when I tried it locally:

```
---- tests::export_genesis_state stdout ----
thread 'tests::export_genesis_state' panicked at 'assertion failed: `(left == right)`
  left: `"307830303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030386532353331346231653431353335393530396230636665663636643430633164376630306662373936376234613863306632313361356266346331303033303331373061326537353937623762376533643834633035333931643133396136326231353765373837383664386330383266323964636634633131313331343030"`,
 right: `"307830303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303039363932323531386235323339616434643731656535386561336562633863336235383763306162376535383935393932333335373034313161633263343365303331373061326537353937623762376533643834633035333931643133396136326231353765373837383664386330383266323964636634633131313331343030"`', node/src/tests.rs:184:5
```

Reply:
Oh yeah, this might be about wasm compilation non-determinism. I suspect that using a raw chainspec would work. Maybe we should write this test against the alphanet spec.

Reply:
Will do, thanks!

Reply:
Seems everything is passing now with alphanet specs, I tried locally and it also works, can you guys try?

Reply:
It's passing for me now (I think), but I'm seeing this:

```
test tests::purge_chain_purges_relay_and_para ... FAILED
test tests::export_current_state ... ok

failures:

---- tests::purge_chain_purges_relay_and_para stdout ----
thread 'tests::purge_chain_purges_relay_and_para' panicked at 'the process should still be running', node/src/tests.rs:70:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    tests::purge_chain_purges_relay_and_para
```

Reply:
hum consistently?

Reply:
For me they pass but its definitely something bad if not everybody is able to work with the same amount of delays

Reply:
Per our call, these tests are all passing now :+1: